src/com/owncloud/android/providers/UsersAndGroupsSearchProvider.java

```diff
@@ -148,31 +156,45 @@ private Cursor searchForUsersOrGroups(Uri uri) {
         if (names.size() > 0) {
             response = new MatrixCursor(COLUMNS);
             Iterator<JSONObject> namesIt = names.iterator();
-            int count = 0;
             JSONObject item;
             String displayName;
+            int count = 0;
             Uri dataUri;
             Uri userBaseUri = new Uri.Builder().scheme("content").authority(DATA_USER).build();
             Uri groupBaseUri = new Uri.Builder().scheme("content").authority(DATA_GROUP).build();
+            Uri remoteBaseUri = new Uri.Builder().scheme("content").authority(DATA_REMOTE).build();
+
             try {
                 while (namesIt.hasNext()) {
                     item = namesIt.next();
                     String userName = item.getString(GetRemoteShareesOperation.PROPERTY_LABEL);
                     JSONObject value = item.getJSONObject(GetRemoteShareesOperation.NODE_VALUE);
-                    byte type = (byte) value.getInt(GetRemoteShareesOperation.PROPERTY_SHARE_TYPE);
+                    int type = value.getInt(GetRemoteShareesOperation.PROPERTY_SHARE_TYPE);
                     String shareWith = value.getString(GetRemoteShareesOperation.PROPERTY_SHARE_WITH);
-                    if (GetRemoteShareesOperation.GROUP_TYPE.equals(type)) {
+
+                    if (ShareType.GROUP.getValue() == type) {
                         displayName = getContext().getString(R.string.share_group_clarification, userName);
                         dataUri = Uri.withAppendedPath(groupBaseUri, shareWith);
+                    } else if (ShareType.FEDERATED.getValue() == type) {
+                        // check if the federated sharing is allowed
+                        FileDataStorageManager manager = new FileDataStorageManager(account, getContext().getContentResolver());
+                        boolean federatedShareAllowed = manager.getCapability(account.name).getFilesSharingFederationOutgoing().isTrue();
```

Comment:
Checking if ```federatedShareAllowed``` should not be done in the ```while``` loop, but before entering it. There are zero chances of it changes between different iterations.

Moreover, probably it may go directly in ```onCreate``` , and ```federatedShareAllowed``` be converted in a member variable.

In both cases, we can rewrite the condition to ```} else if (ShareType.FEDERATED.getValue() = type && federatedShareAllowed)``` , so that we don't need to use a ```continue``` clause (that I'd rather to avoid as a general rule)